client/src/components/SearchInput.js

```diff
@@ -1,16 +1,96 @@
 import React from 'react'
-import { Box, Button, TextInput } from 'grommet'
+import { Box, Button, Select, TextInput, Keyboard } from 'grommet'
+import { useSearchResources } from '../hooks/useSearchResources'
+import { Mappings } from './resources'
+import { sortedObjectKeys } from '../helpers/sortObjectKeys'
+import { getReadable } from '../helpers/readableNames'
+
+export default function SearchInput({
+  initialString = '',
+  onChange,
+  size = 'medium'
+}) {
+  const {
+    query: { search },
+    addFacet,
+    removeFacet,
+    goToSearchResults,
+    setSearchString
+  } = useSearchResources()
+  const [resourceType, setResourceType] = React.useState('ALL')
+  const [string, setString] = React.useState(search || initialString)
```

Comment:
I don't love `string` as a variable name.

Reply:
Could we maybe go with `currentSearchString` or something? It's maybe a bit confusing keeping initialString, string, and search differentiated, so I think more descriptive names might help.

I think I'd also be in favor of changing `initialString` -> `intialSearchString`. I think the concept of "search string" is clear, but if you shorten it down to just `string` it starts to get really ambiguous.

Reply:
`initialString` -> this actually should be deleted and only be populated from the hook
`string` -> `inputValue`
`setString` -> `setinputValue`

Reply:
also changed `search` to `query.search` for clarity

Reply:
and also `setSearchString` is now `setSearchTerm` in the hook